tests/Admin/PoolTest.php

```diff
@@ -245,6 +245,82 @@ public function testGetAdminByAdminCodeForChildInvalidClass(): void
         $this->assertFalse($this->pool->getAdminByAdminCode('sonata.news.admin.post|sonata.news.admin.invalid'));
     }
 
+    /**
+     * @dataProvider getEmptyRootAdminServiceNames
+     */
+    public function testGetAdminByAdminCodeWithInvalidRootCode(string $adminId): void
+    {
+        $adminMock = $this->createMock(AdminInterface::class);
+        $adminMock->expects($this->never())
+            ->method('hasChild');
+
+        $containerMock = $this->createMock(ContainerInterface::class);
+        $containerMock->expects($this->never())
+            ->method('get');
+
+        $poolMock = $this->getMockBuilder(Pool::class)
+            ->setConstructorArgs([$containerMock, 'Sonata', '/path/to/logo.png'])
+            ->disableOriginalClone()
+            ->setMethodsExcept(['getAdminByAdminCode'])
+            ->getMock();
+        $poolMock->expects($this->never())
+            ->method('getInstance');
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Root admin code must contain a valid admin reference, empty string given.');
+        $poolMock->getAdminByAdminCode($adminId);
+    }
+
+    public function getEmptyRootAdminServiceNames()
+    {
+        return [
+            [''],
+            ['   '],
+            ['|sonata.news.admin.child_of_empty_code'],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidChildAdminServiceNames
+     *
+     * @group legacy
+     *
+     * @expectedDeprecation Passing an invalid admin code as argument 1 for Sonata\AdminBundle\Admin\Pool::getAdminByAdminCode() is deprecated since 3.48 and will throw an exception in 4.0.
```

Comment:
```suggestion
     * @expectedDeprecation Passing an invalid admin code as argument 1 for Sonata\AdminBundle\Admin\Pool::getAdminByAdminCode() is deprecated since 3.x and will throw an exception in 4.0.
```